### PR TITLE
Introduce tx command

### PR DESCRIPTION
**Pull request by ipdae**
- Add api for rpc mode activation.
  - get nonce query
  - action command for tx
  - tx command for remote mutation

### CODE REVIEW

File: NineChronicles.Headless.Executable/Commands/TxCommand.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        {
+            _console = console;
+        }
+
+        [Command(Description = "Create new transaction with given actions and dump it.")]
+        public void Sign(
+            [Argument("PRIVATE-KEY", Description = "A hex-encoded private key for signing.")] string privateKey,
+            [Argument("NONCE", Description = "A nonce for new transaction.")] long nonce,
+            [Argument("GENESIS-HASH", Description = "A hex-encoded genesis block hash.")] string genesisHash,
+            [Option("action", new[] { 'a' }, Description = "Hex-encoded actions or a path of the file contained it.")] string[] actions,
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/TxCommand.cs:35]: For consistency:

```suggestion
            [Option("action", new[] { 'a' }, Description = "A hex-encoded actions or a path of the file contained it.")] string[] actions,
```

File: NineChronicles.Headless.Executable/Commands/TxCommand.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                }
+
+                var decoded = (List)Codec.Decode(ByteUtil.ParseHex(a));
+                string type = (Text)decoded[0];
+                Dictionary plainValue = (Dictionary)decoded[1];
+
+                var action = type switch
+                {
+                    nameof(ActivateAccount) => new ActivateAccount(),
+                    _ => throw new CommandExitedException($"Can't determine given action type: {type}", 128),
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/TxCommand.cs:54]:
```suggestion
                    _ => throw new CommandExitedException($"Unsupported action type was passed '{type}'", 128),
```

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs [hunk 3]
```diff
@@ -699,6 +699,91 @@ public async Task Avatar()
...
+            (ActivationKey activationKey, PendingActivationState pendingActivation) =
+                ActivationKey.Create(privateKey, nonce);
+            var pendingActivationStates = new List<PendingActivationState>();
+
+            if (exist)
+            {
+                pendingActivationStates.Add(pendingActivation);
+            }
+
+
```
Comment (moreal) [inline NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs:721]: It is a review for style:

```suggestion

```

File: NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs [hunk 4]
```diff
@@ -699,6 +699,91 @@ public async Task Avatar()
...
+                StaticPeers = ImmutableHashSet<BoundPeer>.Empty
+            };
+
+            var service = new NineChroniclesNodeService(userPrivateKey, properties, null);
+            StandaloneContextFx.NineChroniclesNodeService = service;
+            StandaloneContextFx.BlockChain = service.Swarm?.BlockChain;
+
+            var blockChain = StandaloneContextFx.BlockChain!;
+
+            await blockChain.MineBlock(adminAddress);
```
Comment (moreal) [inline NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs:771]: Is there a reason to mine a block? This test seems not related with the action.

File: NineChronicles.Headless.Executable/Commands/ActionCommand.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        [Command(Description = "Create ActivateAccount action.")]
+        public int ActivateAccount(
+            [Argument("INVITATION-CODE", Description = "A hex-encoded invitation code.")] string invitationCode,
+            [Argument("NONCE", Description = "A hex-encoded nonce for activation.")] string nonceEncoded,
+            [Argument("path", Description = "A file path of hex encoded action.")] string filePath
+        )
+        {
+            try
+            {
+                ActivationKey activationKey = ActivationKey.Decode(invitationCode);
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/ActionCommand.cs:32]: Could you write unittest for the case if the user passed invalid `invitationCode`?

File: NineChronicles.Headless.Executable/Commands/ActionCommand.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                byte[] nonce = ByteUtil.ParseHex(nonceEncoded);
+                Nekoyume.Action.ActivateAccount action = activationKey.CreateActivateAccount(nonce);
+                var encoded = new List(
+                    new[]
+                    {
+                        (Text) nameof(Nekoyume.Action.ActivateAccount),
+                        action.PlainValue
+                    }
+                );
```
[the remark this answers is not in the document]
Comment (ipdae) [reply NineChronicles.Headless.Executable/Commands/ActionCommand.cs:41]: I was use a switch statement to add an action type later, but fix in this pr.

[the remark this answers is not in the document]
Comment (ipdae) [reply NineChronicles.Headless.Executable/Commands/ActionCommand.cs:41]: yes. it is used to switching the type of action inside the tx command.

**Review by area363 — APPROVED** [review]

File: NineChronicles.Headless.Executable/Commands/TxCommand.cs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                privateKey: new PrivateKey(ByteUtil.ParseHex(privateKey)),
+                genesisHash: BlockHash.FromString(genesisHash),
+                timestamp: (timestamp is null) ? default : DateTimeOffset.Parse(timestamp),
+                actions: parsedActions
+            );
+            byte[] raw = tx.Serialize(true);
+
+            if (bytes)
+            {
+                _console.Out.Write(ByteUtil.Hex(raw));
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/TxCommand.cs:72]: By the `bytes` option's description, the `bytes` option expects to print raw bytes, not hexadecimal string.

> Print raw bytes instead of hexadecimal. No trailing LF appended.

Was the above line intended?

Comment (ipdae) [reply NineChronicles.Headless.Executable/Commands/TxCommand.cs:72]: my mistake. i'll fix this.
